### PR TITLE
[MM-47489] Move Shared Channels (Experimental) to Professional

### PR DESCRIPTION
**Pull request by mylonsuren**
#### Summary

Moves Shared Channels to professional license.

#### Ticket Link

https://mattermost.atlassian.net/browse/MM-47489

#### Related Pull Requests

- Has server changes https://github.com/mattermost/mattermost-server/pull/21882

#### Release Note

```release-note
NONE
```

### CODE REVIEW

**Review by AshishDhama — APPROVED** [review]

**Comment by Willyfrog** [comment]
@ashishbhate kind reminder to review when you are feeling better

File: components/admin_console/admin_definition.jsx [hunk 1]
```diff
@@ -6886,7 +6886,10 @@ const AdminDefinition = {
                         help_text: t('admin.experimental.enableSharedChannels.desc'),
                         help_text_default: 'Toggles Shared Channels',
                         help_text_markdown: false,
-                        isHidden: it.not(it.licensedForFeature('SharedChannels')),
+                        isHidden: it.not(it.any(
+                            it.licensedForSku(LicenseSkus.Enterprise),
+                            it.licensedForSku(LicenseSkus.Professional),
+                        )),
```
Comment (ashishbhate) [inline components/admin_console/admin_definition.jsx:6892]: I think we need to check both the feature and the license SKU. I've pinged someone on the growth/self-serve team on the server PR - https://github.com/mattermost/mattermost-server/pull/21882#issuecomment-1362317834

Comment (mylonsuren) [reply components/admin_console/admin_definition.jsx]: After the discussions there, I've also re-added the check for the feature itself, so it should cover all cases now.

**Review by ashishbhate — APPROVED** [review]

**Comment by mylonsuren** [comment]
@neallred This is the associated webapp PR to https://github.com/mattermost/mattermost-server/pull/21882, it mirrors our discussions there, but just wanted to get your eyes on this as well to be safe.

**Review by neallred — APPROVED** [review]
LGTM, thanks @mylonsuren !

**Comment by jgilliam17** [comment]
Thanks @mylonsuren 
The test server (with Enterprise trial) shows a system console option to enable shared channels. I switched between starter, enterprise and professional and both enterprise and professional continue to have the shared channels option.
Screen Shot 2022-12-26 at 11 01 31 AM

**Comment by mylonsuren** [comment]
> Thanks @mylonsuren The test server (with Enterprise trial) shows a system console option to enable shared channels. I switched between starter, enterprise and professional and both enterprise and professional continue to have the shared channels option.


Thanks @jgilliam17, is there anything else blocking this and https://github.com/mattermost/mattermost-server/pull/21882?

**Comment by jgilliam17** [comment]
@mylonsuren no, just one issue I mentioned with enterprise still having shared channels
[E2E report ](https://automation-dashboard.vercel.app/cycle/279952)- no PR related failures.

**Comment by mylonsuren** [comment]
> @mylonsuren no, just one issue I mentioned with enterprise still having shared channels [E2E report ](https://automation-dashboard.vercel.app/cycle/279952)- no PR related failures.

@jgilliam17 Ah sorry I misunderstood your original comment. My understanding is that this feature is being moved down from enterprise to the professional license, but will still be available on both licenses.

@wiersgallak Could you confirm if Shared Channels is being moved exclusively to Professional or will it still be available on Enterprise licenses?

**Comment by kwiersgalla** [comment]
>My understanding is that this feature is being moved down from enterprise to the professional license, but will still be available on both licenses.

@mylonsuren Yes this is correct. Shared Channels will be available in both Professional and Enterprise editions.

**Review by jgilliam17 — APPROVED** [review]
Ah, thanks for clarifying @mylonsuren and @kwiersgalla I misunderstood the ticket description. 
Looks good to merge in that case 🙂

**Pull request merged**